### PR TITLE
Add unit tests for some CSS

### PR DESCRIPTION
**Pull request by rstacruz**
Was about to try and fix the data: URI bugs, but it seems it was already fixed! Anyway, here are some tests.

image

### CODE REVIEW

File: test/utils-tests.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      padding: '0',
+    });
+  });
+
+  it('works for colons', function () {
+    var input = "background: url('data:image/png;base64,abcdef=/+') no-repeat";
+    var output = createStyleJsonFromString(input);
+
+    assert.deepEqual(output, {
+      background: " url('data:image/png;base64,abcdef=/+') no-repeat"
```
Comment (rstacruz) [inline test/utils-tests.js]: maybe we should consider trimming spaces where approriate?

**Comment by rstacruz** [comment]
Matter of personal preference really, I just thought it'd make a lot of sense to have unit tests as well as integration tests.

I think it'd be valuable in iterating certain parts of the app. For instance, if we're to refactor `margin: 0` parsing into `{ margin: '0' }` instead of `{ margin: ' 0' }` (with leading space), it'd be easier if we can just write test cases into the unit test here, rather than the full suite tests in html-to-react-tests.js.

It's also easier to write unit tests (compared to integration tests) to cover all edge cases comprehensively, like say:

- `margin: 0`
- `margin: 0;`
- `margin:0;padding:0;`
- `background:url('http://foo.com/image.png');`
- `background:url("http://foo.com/image:x.png?a=1&b=4")`

They can be written as integration tests, but if it fails, you wouldn't easily know what part of the library is failing.

These tests may not be necessary now, but if there will be a big refactoring down the road, it might help.

...anyway, it's all a matter of personal preference, really. I love unit tests, even for something with 100% coverage, for the reasons above, but not everyone may feel the same. I just ended up writing these in preparation for a PR (which ended up not being needed), so I figured I'd put it out there if you'd like to merge it :)

**Comment by rstacruz** [comment]
There's also a small argument to be made for how the tests depend on `ReactDOMServer` — if their implementation changes in the future (eg, start using `'` instead of `"` by default), many tests would fail without any clear indication of where the failure is.

**Comment by aknuds1** [comment]
I'm of the philosophy that tests shouldn't depend on implementation
details, it makes them liable to break in case you change implementation
without breaking actual functionality, a no-no in my book as it makes
future maintenance difficult for no good reason.

Nevertheless I'll have a look at your changes later to see if there's any
real benefit.

**Comment by aknuds1** [comment]
Sorry, I don't see the point of adding unit tests, especially considering how tiny the library is. I also don't like that the interface of the utils module gets enlargened just for the sake of tests, it should stay as slim as possible.
